platforms/jvm/jvm-services/src/main/java/org/gradle/jvm/toolchain/internal/EnvironmentVariableJavaHomeInstallationSupplier.java

```diff
@@ -38,6 +35,7 @@ public String getSourceName() {
 
     @Override
     public Set<InstallationLocation> get() {
+        String javaHomePath = buildOptions.getEnvironmentVariableValue("JAVA_HOME");
```

Comment:
The first version was doing this in the constructor, but that means that simply initializing the toolchain system would make the configuration cache entry depend on `JAVA_HOME`. It should do so only if toolchains are used.

Reply:
This is a good catch, in fact we follow the same also on `EnvironmentVariableListInstallationSupplier` for the same reason